src/hooks/useCreateGQLClient/index.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
+import { HttpLink, ApolloClient, NormalizedCacheObject } from "@apollo/client";
+import { useAuthDispatchContext } from "context/Auth";
+import { cache } from "gql/client/cache";
+import {
+  authenticateIfSuccessfulLink,
+  authLink,
+  logGQLToSentryLink,
+  logErrorsLink,
+  retryLink,
+} from "gql/client/link";
+import { secretFieldsReq } from "gql/fetch";
+import { environmentVariables } from "utils";
+import { leaveBreadcrumb, SentryBreadcrumb } from "utils/errorReporting";
+import { fetchWithRetry, shouldLogoutAndRedirect } from "utils/request";
+
+const { getGQLUrl } = environmentVariables;
+
+export const useCreateGQLCLient = (): ApolloClient<NormalizedCacheObject> => {
+  const { dispatchAuthenticated, logoutAndRedirect } = useAuthDispatchContext();
+  const [secretFields, setSecretFields] = useState<string[]>();
+  const [gqlClient, setGQLClient] = useState<any>();
+
+  useEffect(() => {
+    fetchWithRetry(getGQLUrl(), secretFieldsReq)
```

Comment:
With the above/below? suggestion you can add some type safety here by using the generic 
```suggestion
    fetchWithRetry<SecretFieldsQuery>(getGQLUrl(), secretFieldsReq)
```